main.py

```diff
@@ -138,13 +149,14 @@ async def get_task_status(task_id: str) -> TaskStatus:
     return tasks[task_id]
 
 
-@app.get("/exceptions/{task_id}")
+@app.get("/exceptions/{task_id}", response_model=list[CompilerErrorModel])
 async def get_task_exceptions(task_id: str) -> dict:
```

Comment:
If you use a typing annotation for return, FastAPI will automatically parse that into JSON using the model you give it (even inside a `list`)
```suggestion
@app.get("/exceptions/{task_id}")
async def get_task_exceptions(task_id: str) -> list[CompilerErrorModel]:
```